django/db/backends/mysql/base.py

```diff
@@ -442,3 +450,7 @@ def mysql_is_mariadb(self):
     def sql_mode(self):
         sql_mode = self.mysql_server_data["sql_mode"]
         return set(sql_mode.split(",") if sql_mode else ())
+
+    @cached_property
+    def has_db_uuid_type(self):
+        return self.mysql_is_mariadb and self.mysql_version >= (10, 7, 0)
```

Comment:
@felixxm Thanks for comments and code improvement suggestion, looks nice.
I introduced `has_db_uuid_type` because I met an issue and wanted differentiate two cases. One is when db has buildin UUID type but library doesn't support conversion to python uuid (mysqlclient and we can meet in future in other dbs) and second, db has buildin UUID type and library supports conversion to uuid type (psycopg). For me `has_native_uuid_field` indicates second case. In this PR it is enough to have only `has_native_uuid_field` and if I won't need that in future I can refactor code as you suggested.